launcher/agent/agent.go

```diff
@@ -79,14 +91,27 @@ func (a *agent) Attest(ctx context.Context) ([]byte, error) {
 		return nil, err
 	}
 
-	resp, err := a.client.VerifyAttestation(ctx, verifier.VerifyAttestationRequest{
+	req := verifier.VerifyAttestationRequest{
 		Challenge:      challenge,
 		GcpCredentials: principalTokens,
 		Attestation:    attestation,
-	})
+	}
+
+	if a.launchSpec.Experiments.EnableSignedContainerImage {
+		signatures := a.sigsFetcher(a.launchSpec.SignedImageRepos)
+		if len(signatures) > 0 {
+			req.ContainerImageSignatures = signatures
+			a.logger.Printf("Found container image signatures: %v\n", signatures)
+		}
+	}
+
+	resp, err := a.client.VerifyAttestation(ctx, req)
```

Comment:
Can you add unit test coverage here? The fake verifier client should expect to see signatures.